lib/PuppeteerSharp/ConnectOptions.cs

```diff
@@ -48,8 +48,14 @@ public class ConnectOptions : IBrowserOptions, IConnectionOptions
         /// Optional factory for <see cref="WebSocket"/> implementations.
         /// If <see cref="Transport"/> is set this property will be ignored.
         /// </summary>
+        [Obsolete("Use TransportFactory instead")]
         public Func<Uri, IConnectionOptions, CancellationToken, Task<WebSocket>> WebSocketFactory { get; set; }
```

Comment:
I think we shouldn't mark this as `Obsolete`. We should still be able to do
```cs
WebSocketFactory = (uri, socketOptions, cancellationToken) =>
	{
		return System.Net.WebSockets.SystemClientWebSocket.ConnectAsync(uri, cancellationToken);
	},
```

Reply:
Ok, I will submit a different pull-request with a consolidation of the changes in that case.